crates/bevy_asset/src/diagnostic/asset_count_diagnostics_plugin.rs

```diff
@@ -48,6 +48,6 @@ impl<T: Asset> AssetCountDiagnosticsPlugin<T> {
     }
 
     pub fn diagnostic_system(mut diagnostics: ResMut<Diagnostics>, assets: Res<Assets<T>>) {
-        diagnostics.add_measurement(Self::diagnostic_id(), assets.len() as f64);
+        diagnostics.add_measurement(Self::diagnostic_id(), || assets.len() as f64);
```

Comment:
Why do we need this zero-argument closure now?

Reply:
to be able to not compute the value if the diagnostic is not enabled